TF-14957 Add Team Tokens Index API (#667)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/changelog.mdx

Keep track of changes to the API for HCP Terraform and Terraform Enterprise.

## 2024-06-20

> * Add documentation for new [team tokens](/terraform/cloud-docs/api-docs/organization-team-tokens) API that lists team tokens in an organization.

Reviewer: ```suggestion
* Add documentation for new API endpoint that lists an [organization's team tokens](/terraform/cloud-docs/api-docs/organization-team-tokens).
```

* Add reference to organization team token api on teams, index and navigation page.

> * Add a `sort` parameter to the [Team Tokens list API](/terraform/cloud-docs/api-docs/organization-team-tokens#query-parameters) to allow sorting tokens by team name, created by, expired at and last used.

Reviewer: I also don't think we need this bullet point because it's covered by the first bullet point. 
```suggestion
```

Keep track of changes to the API for HCP Terraform and Terraform Enterprise.

## 2024-06-28

> * Add documentation for new API endpoint that lists an [organization's team tokens](/terraform/cloud-docs/api-docs/team-tokens).

Reviewer: ```suggestion
* Add documentation for a new API endpoint that lists an [organization's team tokens](/terraform/cloud-docs/api-docs/team-tokens).
```
